### ISSUE TITLE
script doing nothing

### ISSUE THREAD

**Issue by andruccho** [OpenNebula/addon-image-backup#8]
It outputs momentally:

Backup images
Backup deployments - system datastores without non-persistent disks...

Exiting with no errors and nothing really backed up.
What am I doing wrong? 

Opennebula 5.10

**Comment by feldsam** [comment]
How you are running script?

**Comment by richman1000000** [comment]
@andruccho I had exact same issue.
you need to use QCOW2 datastore type. 
![image](https://user-images.githubusercontent.com/9943374/80805203-14f7fe80-8bc0-11ea-8527-0af6dcb3399b.png)

**Comment by feldsam** [comment]
No reply, closing

**Comment by andruccho** [comment]
> How you are running script?

Sorry for late reply, I was out of work

Well, I'm just ran
./one-image-backup.js -vDs
and getting this with no real work
`Backup deployments - system datastores without non-persistent disks...
`
or backup.sh without parameters and getting this with no real work
```
Backup images
Backup deployments - system datastores without non-persistent disks...
```

My config file is as follow: 

    address: 'http://localhost:2633/RPC2',
    user: 'oneadmin',
    token: 'token_from_#settings-tab'
    backupDir: '/4tb/one_backup/',
    backupTmpDir: '/4tb/one_backup_temp/',
    bridgeList: ['off.loc'],

off.loc is my opennebula server, and also it is a one single node. I ran scripts on it.
Folders backupDir and backupTmpDir were created before running.

**Comment by feldsam** [comment]
Hi @andruccho, post here you datastores templates. Did you check what @richman1000000 wrote? This script only backups datastores with transfer manager == qcow2 and only persistent images.

**Comment by andruccho** [comment]
SYSTEM_DS
<img width="444" alt="Screenshot 2020-05-15 at 11 19 10" src="https://user-images.githubusercontent.com/23392714/82028260-0f111b80-969e-11ea-9185-be22b776cbc5.png">

IMAGE_DS
<img width="444" alt="Screenshot 2020-05-15 at 11 19 36" src="https://user-images.githubusercontent.com/23392714/82028293-1cc6a100-969e-11ea-93e4-2561ccb74133.png">

**Comment by feldsam** [comment]
hmm, I don't know why. Hard to debug remotely. You should debug what XML-RPC api is returning. If there are data

**Comment by feldsam** [comment]
On line 107 add `console.log(results)` and save to pastebin.com and add link here

**Comment by andruccho** [comment]
now there's 
`{ datastores: undefined }
`
appeared in output

tried 
`./one-image-backup.js` -a 112 -vDs`
but the same

**Comment by richman1000000** [comment]
did you CREATED new datastore with qcow2?
If you tried to change existing datastore , this is big  mistake!!

you need to create a new datastore and during creation choose qcow2 
![image](https://user-images.githubusercontent.com/9943374/82057585-3f6fae80-96cc-11ea-8264-987dc0f927d5.png)

**Comment by andruccho** [comment]
i think yes, i created it as qcow2 initially
just tested with brand new qcow2 datastore -- all the same

**Comment by feldsam** [comment]
there is some problem getting data from api.

on line 107 add `console.log(err, results);`

**Comment by andruccho** [comment]
`Error: [one.datastorepool.info] User couldn't be authenticated, aborting call.
    at /root/addon-image-backup/node_modules/opennebula/lib/modem.js:32:23
    at Deserializer.callback (/root/addon-image-backup/node_modules/xmlrpc/lib/deserializer.js:41:7)
    at Deserializer.onDone (/root/addon-image-backup/node_modules/xmlrpc/lib/deserializer.js:91:12)
    at emitNone (events.js:86:13)
    at SAXStream.emit (events.js:185:7)
    at Object.SAXStream._parser.onend (/root/addon-image-backup/node_modules/xmlrpc/node_modules/sax/lib/sax.js:171:8)
    at emit (/root/addon-image-backup/node_modules/xmlrpc/node_modules/sax/lib/sax.js:325:33)
    at end (/root/addon-image-backup/node_modules/xmlrpc/node_modules/sax/lib/sax.js:363:3)
    at Object.end (/root/addon-image-backup/node_modules/xmlrpc/node_modules/sax/lib/sax.js:137:24)
    at SAXStream.end (/root/addon-image-backup/node_modules/xmlrpc/node_modules/sax/lib/sax.js:209:16) { datastores: undefined }
Backup deployments - system datastores without non-persistent disks...
`

**Comment by feldsam** [comment]
use just `username:password`

**Comment by andruccho** [comment]
yep! now it works as expected
script doesn't work from oneadmin user (I didn't try to realize why), so I ran it from root. it was needed to add root's public key to oneadmin's authorized_keys. Also, oneadmin has to be an owner of backup temp dir. 

To live backup of running VMs, you should have QEMU 2.9 at least, and latest QEMU guest agents installed on your VMs. 

Also, Windows systems lower than 8 do not support guest-fstrim QEMU feature, and therefore cannot be live-backuped.

Thank you @feldsam for your support!

**Comment by feldsam** [comment]
Hello, thanks for the info.

- `qemu 2.9` - I updated readme, I forgot about that, we were using `qemu-kvm-ev` repo on centos
- `fstrim` - there is config var `libvirtUseDomFsTrim`, so you can enable/disable this feature. Anyway, if it fails, it just print error and continue doing live backup. So you can leave it enabled.

**Comment by feldsam** [comment]
I just released version 1.12.1. I added error handling, so user can see error you hit + updated qemu version in readme
